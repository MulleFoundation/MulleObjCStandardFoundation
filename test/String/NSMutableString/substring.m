//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
//#import "MulleStandaloneObjCFoundation.h"


   static unichar  _UTF32Unichar[] = { 0x0001f463, 0x00000023, 'A', 'B', 'C', 0x000020ac, 0x0001f3b2 };   /* UTF32 feet, hash, euro, dice */

int main(int argc, const char * argv[])
{
   NSMutableString   *s;
   NSString          *other;
   NSMutableString   *clone;
   NSCharacterSet    *charSet;
   NSString          *strings[ 2];
   NSUInteger        i;

   s = [NSMutableString string];

   [s appendString:@"VfL"];
   [s appendString:@" "];
   [s appendString:@"Bochum"];
   [s appendString:@" "];
   [s appendString:@"1848"];
   [s appendString:@" "];
   [s appendString:[[[NSString alloc] initWithUTF8String:"m\303\266hp"] autorelease]];
   [s appendString:@" "];
   [s appendString:[[[NSString alloc] initWithCharacters:_UTF32Unichar
                                                  length:7] autorelease]];
   other = [s substringWithRange:NSRangeMake( 2, 10)];
   printf( "%s\n", [other UTF8String]);

   other = [s substringWithRange:NSRangeMake( 0, 0)];
   printf( "%s\n", [other UTF8String]);

   for( i = [s length] / 2; i; i--)
   {
      if( i * 2 <= [s length])
      {
         other = [s substringWithRange:NSRangeMake( i, [s length] - i * 2)];
         printf( "%s\n", [other UTF8String]);
      }
   }
   return( 0);
}
