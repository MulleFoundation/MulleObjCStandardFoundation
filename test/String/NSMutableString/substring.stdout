L Bochum 1


48
848 
1848 m
 1848 mÃ¶
m 1848 mÃ¶h
um 1848 mÃ¶hp
hum 1848 mÃ¶hp 
chum 1848 mÃ¶hp ğŸ‘£
ochum 1848 mÃ¶hp ğŸ‘£#
Bochum 1848 mÃ¶hp ğŸ‘£#A
 Bochum 1848 mÃ¶hp ğŸ‘£#AB
L Bochum 1848 mÃ¶hp ğŸ‘£#ABC
fL Bochum 1848 mÃ¶hp ğŸ‘£#ABCâ‚¬
